### DISCUSSION TITLE
[x] Defining the Research Software Engineer: Skills, Requirements, and Pathways

### DISCUSSION THREAD

**Discussion by richarddushime** [RSEAA/RSEAA.github.io#44] [Discussion topics for #RSEAA2023]
This discussion aims to delve into the multifaceted role of Research Software Engineers (RSEs), shedding light on the core skills, essential requirements, and diverse pathways that define this dynamic profession. As RSEs bridge the gap between research and technology, join us to explore the evolving landscape of expertise, the journey to becoming an RSE, and the versatile skillset that underpins their invaluable contributions to academia and beyond.

**Comment by nrmay** [comment]
At a minimum, all RSEs should have a basic understanding of:

- how to build and run software (inc. code) 
- the research lifecycle and research domains

The relative expertise in these two aspects vary depending on the pathway of the individual into the RSE domain.

**Comment by betsybookwyrm** [comment]
My notes:

- what defines an RSE?
	- most people don't realise they're an RSE
	- common to not feel like doing proper research
	- in-betweenness
	- some are domain-specific, some are technical skills-specific (academic/professional background divide?)
- key skills
	- writing code that's usable by other people
	- abstraction
	- problem solving
- pathways
	- researcher who found themselves doing increasing amounts of coding and realising they need to upskill in software engineering
	- software professional coming to work in a research environment not understanding research methods and contexts
- moving from being seen as a technician to being a facilitator of research outcomes - being seen as an intellectual contributor
	- "academic specialist" roles
	- emerging and shifting concept of research infrastructure
	- guidelines on when research support becomes research collaboration (i.e. acknowledgement vs authorship)
- careers
	- it's easy to get trapped out of academic career pathways, difficult to move between academia and industry
	- recognition is what is most lacking at the moment
		- specialist journals are one thing but not enough
	- RSE as a concept doesn't exist in Japan, at least outside of specific groups
	- need to standardise (or at least clarify) expectations of RSEs around things like publishing, software support & maintenance
- lack of place for ongoing work
	- testing and maintenance are particularly unrecognised labour with software
	- community development and management of software doesn't fit in anywhere
	- Software doesn't go away when the funding runs out. There are obligations relating to reproducibility, to commercialisation, to public engagement, and others which often require keeping the software going. (from Philip Dart)

**Comment by rowlandm** [comment]
RSE intellecual contribution as opposed to an easily "replacable" kind of "technical" person.

**Comment by rowlandm** [comment]
What is the top of the RSE career path?

**Reply by abickerstaffe** [reply]
Self-sufficient funding, project revenue, probably to fund a team.

**Reply by rowlandm** [reply]
I never thought about it that way before!

**Reply by kozo2** [reply]
Is there anyone who can be the RSE **model**?
Or is there still no one who can?

**Reply by dwannetts** [reply]
In CSIRO, the upper levels are responsible for directing and initiating policy & research directions.  These directions in turn start projects which bring in funding, etc,.

**Reply by betsybookwyrm** [reply]
tbh as a professional staff member all I want right now is permanent employment rather than fixed term. I know many/all of you academics here will be in the same situation! Any other sort of career progression is less of a concern to me at the moment.

**Reply by rowlandm** [reply]
@kozo2 - here are some people based on the fact they have reached a professorship while doing RSE type things. I think only Simon Hettrick is considered an RSE professor.

Professor Douglas Boyle https://findanexpert.unimelb.edu.au/profile/142641-dougie-boyle

Professor Andy Hogg https://earthsciences.anu.edu.au/people/academics/professor-andy-hogg

Professor Kim-Anh Le Cao https://findanexpert.unimelb.edu.au/profile/791255-kim-anh-le-cao

Professor Tony Papenfuss https://www.wehi.edu.au/researcher/tony-papenfuss/

Professor Simon Hettrick https://society-rse.org/about/governance/simon-hettrick/

**Reply by kozo2** [reply]
@rowlandm Is there a pathway to be the top of the RSE without becoming a professor?

**Reply by pr4deepr** [reply]
Thanks for linking to this topic @kozo2 . 
Good point @betsybookwyrm. It seems like you have to choose between stability or career progression. There isn't much of a happy medium. 
In WEHI, there is growing recognition for a career progression pathway for technology scientists and they're trying to change the metrics..

**Comment by abickerstaffe** [comment]
Ability to translate domain specific details into abstract concepts - even if the researchers are not so interested in the abstract concepts.

Making sure your work is recognised as key research enabling contribution. An intellectual contribution that can shape research methods!

**Comment by pr4deepr** [comment]
Is there enough recognition in academia to attract talent from industry?

**Reply by pr4deepr** [reply]
Need better monetary support and career support

**Reply by betsybookwyrm** [reply]
imo the fixed-term/casual contracts are the biggest barrier. People (like me) come to research support from tech because we want to help make the world better and we've come to realise that is a hard task in the tech industry. But the lack of stability is a huge disincentive.

In Australia, we'll see if the Fair Work changes will help the situation in a few months' time. But university execs will try their hardest to maintain the status quo. We'll see what happens.

**Reply by kiowa-scott-hurley** [reply]
I've managed to get stability by getting into the public service. 

The cost of this decision has been: 

-  The well reported pay discrepancies between APS and industry (where proposing below-inflation pay-increases are scandalous);
- The limited ability to share with the broader RSE/HPC community as a Defence staff member;
-  The lack of career structures for technical or research staff (this isn't unique to the APS...) At the tender age of 25, any future promotions for me will mean leading a team. While I love the idea of leading others, it's certainly a different skillset and very different work than benchmarking software or writing code.

**Comment by abickerstaffe** [comment]
This year my Informatics Team created a formalised Informatics Support Agreement document, which outlines how we can support research. It defines two types of support -- as collaborators, or support as a service -- each with different costing models. We cost to incentivise the collaborative approach, because we want to be involved from the inception of research projects, including methods, writing the grant, publications, seminars, investigator meetings, etc. But support as a service can also work for smaller, one-off jobs, or for consultations that might just be a 2-5 hours long.

**Comment by nrmay** [comment]
There is a broad framework, called SFIA, that classifies IT related skills and how they are applied at different levels of responsibility. This can be used to define roles in the IT domain and could be applied to RSE roles and pathways.
https://sfia-online.org/en/sfia-8/sfia-views/full-framework-view/

**Comment by rowlandm** [comment]
Digital humanities is only a "thing" in pockets. This doesn't make it easy for RSEs in the humanities.

**Comment by rowlandm** [comment]
No one wants to talk about the plumbing and maintenance around software.

**Reply by abickerstaffe** [reply]
They want it, just don't want to pay for it ;) because it's not a glamorous thing. Doesn't result in new features you can see. But I often highlight the importance from the security angle, which gets people to pay attention.

**Reply by manodeep** [reply]
Yup! I recently had to use a piece of software meticulously created by an RSE ~4 years ago but unmaintained since then - basically impossible to use! Software rot is for real

**Comment by rowlandm** [comment]
Software can also be difficult to publish as a paper as journals often require some threshold of ‘novelty’. Often we may just need to rewrite/reimplement/make more robust some non-novel method, but this often isn’t enough for publication.

**Comment by rowlandm** [comment]
Or the hard work involved in taking a method (computational or otherwise) that's common in one discipline and introducing it to another
 
Software doesn't go away when the funding runs out.  There are obligations relating to reproducibility, to commercialisation, to public engagement, and others which often require keeping the software going.

**Comment by rowlandm** [comment]
Don't know how to be a RSE in Japan.

**Reply by kozo2** [reply]
In the Zoom discussion, it was mentioned that there are some researchers in biomedical image analysis in Japan as well, but I interpret researchers and RSEs as different things. How should we define the difference between researchers and RSEs?

**Reply by pr4deepr** [reply]
You are right that they are different, but I think the boundary is not always clear.
I think an RSE can be a researcher and a researcher can be an RSE.

**Reply by kozo2** [reply]
Thank you for the reply.
Indeed, that boundary may not be clear.
In the end, I've come to reconsider that it might also depend on the employer (of the RSE).

**Comment by chesadi** [comment]
some career paths are available based on their qualifications

**Comment by pr4deepr** [comment]
Cultural differences between domains; some domains are well integrated. 
some are seen as plumbers..
Differences in behaviours. 

At an institution level, need to say what they expect from RSEs: need to publish, software maintanence & support? Metrics are different for RSEs. What are we being benchmarked against?

Formalising the process.

**Comment by rowlandm** [comment]
Idea: Bring together RSE teams to talk about collaborative models and transactional models of working with researchers.

**Reply by pr4deepr** [reply]
COuld collate how existing agreements look like for different groups...

**Reply by rowlandm** [reply]
including SLAs (service level agreements) for some of the things people have built

**Reply by betsybookwyrm** [reply]
My team has thought about this issue *a lot*. I'd love to be part of that conversation (although if possible I'd get my manager to do it in my place because she's thought about it even more than I have)

**Comment by pr4deepr** [comment]
Define what 'top' looks like for RSEs. If we can know that, we can define the pathways and skills required.

**Reply by kozo2** [reply]
I agree.
It's also related to https://github.com/RSEAA/RSEAA.github.io/discussions/44#discussioncomment-6985694 ,
I don't feel that becoming a professor/researcher means the 'top'.

**Reply by manodeep** [reply]
Agree with @pr4deepr - not really sure what "top" means for an RSE in the research sector. For Australian Universities, the RSEs typically get a HEW role. As long as organisations keep this distinction between Academic and Research Support personnel (which in turn relates to valuing the research outcome, i.e., a research paper, and *not* the research process), we are going to be stuck with ways of trying to fit a square peg into a round hole. From what I can tell, as soon as you get a HEW role, your career stalls, and the way to move up is by moving elsewhere. 

Pragmatically speaking though this distinction is likely to be around for a bit longer.
